### PR TITLE
CLN: Remove is_null_period

### PR DESCRIPTION
**Pull request by dsaxton**
is_null_period seems only to be an alias for checknull_with_nat so I think it can be removed?

### CODE REVIEW

File: pandas/_libs/missing.pyx [hunk 1]
```diff
@@ -40,6 +40,7 @@ cpdef bint checknull(object val):
      - NaT
      - np.datetime64 representation of NaT
      - np.timedelta64 representation of NaT
+     - NA
```
Comment (jbrockmendel) [inline pandas/_libs/missing.pyx:43]: this is an unrelated cleanup?

**Comment by jbrockmendel** [comment]
LGTM

File: pandas/_libs/missing.pyx (hunk 1 above)
Comment (dsaxton) [reply pandas/_libs/missing.pyx:43]: Yeah I just noticed that was missing from the docstring so added it here

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Thanks!
